src/chapters/unit-testing/tdd.rst

```diff
@@ -0,0 +1,102 @@
+Test-Driven Development
+=======================
+We all want to write code that *works*! How do you know if your code works? You can
+manually run your code and review a list of requirements to ensure all are covered.
+Or you could write a unit test for each requirement.
+
+.. index:: ! TDD
+
+.. index::
+   single: TDD; test-driven development
+
+.. index::
+   single: unit testing; TDD
+
+**Test-driven development (TDD)** is a software development process where every feature is
```

Comment:
Drive home the *driven* part of TDD by using "before" or "first" to make it more explicit.